From e54f6b935b595d813d558620b083a9bf40022543 Mon Sep 17 00:00:00 2001
From: Robert Nagy <robert@openbsd.org>
Date: Sat, 1 Oct 2016 11:23:40 +0200
Subject: [PATCH 51/52] first_run: barf if metrics_reporting is activated
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This patch is for debugging purposes and is meant for discovering
whether  despite our patching efforts  metrics_reporting is still
enabled somehow, and if so, say so on stderr.

Result: As of October 2014, metrics_reporting seems successfully
disabled, as no message is emitted.
---
 chrome/browser/first_run/first_run_internal_posix.cc | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/chrome/browser/first_run/first_run_internal_posix.cc b/chrome/browser/first_run/first_run_internal_posix.cc
index 0e43c63..2abfeb7 100644
--- a/chrome/browser/first_run/first_run_internal_posix.cc
+++ b/chrome/browser/first_run/first_run_internal_posix.cc
@@ -19,6 +19,7 @@
 #include "components/metrics/metrics_reporting_default_state.h"
 #include "components/prefs/pref_service.h"
 #include "components/startup_metric_utils/browser/startup_metric_utils.h"
+#include <cstdio>
 
 namespace first_run {
 namespace internal {
@@ -33,6 +34,9 @@ void DoPostImportPlatformSpecificTasks(Profile* profile) {
   if (internal::IsOrganicFirstRun() && !local_state_file_exists) {
     if (ShowFirstRunDialog(profile)) {
       bool is_opt_in = first_run::IsMetricsReportingOptIn();
+      if (is_opt_in) {
+        fprintf(stderr, "*** metrics_reporting = 1\n");
+      }
       metrics::RecordMetricsReportingDefaultState(
           g_browser_process->local_state(),
           is_opt_in ? metrics::EnableMetricsDefault::OPT_IN
-- 
2.6.6

